pkg/envutil/envutil_test.go

```diff
@@ -0,0 +1,258 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package envutil
+
+import (
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestMatchesPattern(t *testing.T) {
+	tests := []struct {
+		name     string
+		pattern  string
+		expected bool
+	}{
+		{"PATH", "PATH", true},
+		{"PATH", "HOME", false},
+		{"SSH_AUTH_SOCK", "SSH_*", true},
+		{"SSH_AGENT_PID", "SSH_*", true},
+		{"HOME", "SSH_*", false},
+		{"XDG_CONFIG_HOME", "XDG_*", true},
+		{"_LIMA_TEST", "_*", true},
+		{"LIMA_HOME", "_*", false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name+"_matches_"+tt.pattern, func(t *testing.T) {
+			result := matchesPattern(tt.name, tt.pattern)
+			if result != tt.expected {
+				t.Errorf("matchesPattern(%q, %q) = %v, want %v", tt.name, tt.pattern, result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestMatchesAnyPattern(t *testing.T) {
+	patterns := []string{"PATH", "SSH_*", "XDG_*"}
+
+	tests := []struct {
+		name     string
+		expected bool
+	}{
+		{"PATH", true},
+		{"HOME", false},
+		{"SSH_AUTH_SOCK", true},
+		{"XDG_CONFIG_HOME", true},
+		{"USER", false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := matchesAnyPattern(tt.name, patterns)
+			if result != tt.expected {
+				t.Errorf("matchesAnyPattern(%q, %v) = %v, want %v", tt.name, patterns, result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestParseEnvList(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected []string
+	}{
+		{"", nil},
+		{"PATH", []string{"PATH"}},
+		{"PATH,HOME", []string{"PATH", "HOME"}},
+		{"PATH, HOME , USER", []string{"PATH", "HOME", "USER"}},
+		{" , , ", []string{}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.input, func(t *testing.T) {
+			result := parseEnvList(tt.input)
+			if !reflect.DeepEqual(result, tt.expected) {
+				t.Errorf("parseEnvList(%q) = %v, want %v", tt.input, result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestGetBlockAndAllowLists(t *testing.T) {
+	originalBlock := os.Getenv("LIMA_SHELLENV_BLOCK")
+	originalAllow := os.Getenv("LIMA_SHELLENV_ALLOW")
+	defer func() {
+		if originalBlock != "" {
+			t.Setenv("LIMA_SHELLENV_BLOCK", originalBlock)
+		} else {
+			os.Unsetenv("LIMA_SHELLENV_BLOCK")
+		}
+		if originalAllow != "" {
+			t.Setenv("LIMA_SHELLENV_ALLOW", originalAllow)
+		} else {
+			os.Unsetenv("LIMA_SHELLENV_ALLOW")
+		}
+	}()
+
+	t.Run("default config", func(t *testing.T) {
+		os.Unsetenv("LIMA_SHELLENV_BLOCK")
+		os.Unsetenv("LIMA_SHELLENV_ALLOW")
+
```

Comment:
When I removed all `os.Unsetenv` calls in this file and ran the tests, they still passed. So I think we can remove `os.Unsetenv` here and below, since `t.Setenv` automatically unsets the environment variables.

```suggestion
```